src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpPreferStreamAsyncMemoryOverloads.Fixer.cs

```diff
@@ -51,5 +51,23 @@ protected override bool IsSystemNamespaceImported(IReadOnlyList<SyntaxNode> impo
             }
             return false;
         }
+
+        protected override bool IsPassingZeroAndBufferLength(SyntaxNode bufferValueNode, SyntaxNode offsetValueNode, SyntaxNode countValueNode)
+        {
+            return
+                // First argument should be an identifier name node
+                bufferValueNode is IdentifierNameSyntax firstArgumentIdentifierName &&
+                // Second argument should be a literal expression node...
+                offsetValueNode is LiteralExpressionSyntax secondArgumentLiteralExpression &&
+                // and its value should be zero
+                secondArgumentLiteralExpression.Token.ValueText == "0" &&
+                // Third argument should be a member access node...
+                countValueNode is MemberAccessExpressionSyntax thirdArgumentMemberAccessExpression &&
+                thirdArgumentMemberAccessExpression.Expression is IdentifierNameSyntax thirdArgumentIdentifierName &&
+                // whose identifier is that of the first argument...
+                firstArgumentIdentifierName.Identifier.ValueText == thirdArgumentIdentifierName.Identifier.ValueText &&
```

Comment:
Similarly, is this the best way to verify that both the `buffer` argument and the `buffer.Length` argument are the same variable?

Reply:
I think this is the best possible heuristic..